src/core/core.scale.js

```diff
@@ -1183,6 +1183,19 @@ var Scale = Element.extend({
 				me._drawLabels.apply(me, arguments);
 			}
 		}];
+	},
+
+	/**
+	 * @private
+	 */
+	_getMatchingVisibleMetas: function(type) {
```

Comment:
Is placing this on the scale class the best idea?

Reply:
Not quite sure. What cons are you thinking of?

Reply:
Mostly concerns about adding more stuff to the scale class. It's already fairly large. If this made the code simplest though, I'm happy to stick with it